spec/requests/access_tokens_spec.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'AccessTokens', type: :request do
+  describe 'POST /signin' do
+    let(:user) { FactoryBot.create :user }
+    let(:grant_type) { 'password' }
+    let(:username) { user.email }
+    let(:password) { user.password }
+
+    context '正当なパラメータを渡したとき' do
+      subject do
+        post signin_path, params: { grant_type:, username:, password: }
+        response
+      end
+
+      it { is_expected.to have_http_status :success }
+      it { is_expected.to have_attributes media_type: 'application/json' }
+
+      it '値がno-storeであるCache-Controlヘッダを含む' do
+        expect(subject.get_header('Cache-Control')).to eq 'no-store'
+      end
+
+      it '値がno-cacheであるPragmaヘッダを含む' do
+        expect(subject.get_header('Pragma')).to eq 'no-cache'
+      end
+
+      it 'レスポンスボディに発行したアクセストークンとそのメタデータを含む' do
+        expect(subject.parsed_body).to include(
+          'access_token' => user.access_tokens.last.token,
+          'token_type'   => 'bearer',
+          'expires_in'   => 1.hour,
+          'scope'        => 'READ WRITE'
+        )
+      end
+
+      context 'パラメータにscopeが含まれているとき' do
+        subject do
+          post signin_path, params: { grant_type:, username:, password:, scope: }
```

Comment:
subject をいくつも書きたくないので、僕ならこう書きますね。
```suggestion
          post signin_path, params: { grant_type:, username:, password:, scope: }.compact
```

こう書いておくと `scope` 変数にどういう値をセットするかで、パラメータを送信するのかしないのかを制御できます (subject の定義は１つだけで済む)

Reply:
へー、こんなメソッドがあるんですね。

https://docs.ruby-lang.org/ja/latest/method/Hash/i/compact.html

Reply:
Array や Enumerable にも存在するので、ちょこちょこ使いますね。
この手のメソッドは徐々に手持ちのカードを増やしておくと便利なので、都度覚えていくと良さそうです。